### ISSUE TITLE
Introduce flag to set all frozen modules to `eval` mode

### ISSUE THREAD

**Issue by moritzhauschulz** [ecmwf/WeatherGenerator#1687]
### Describe the task. It can be a feature, documentation, etc.

The model currently allows to specify modules to freeze via the `freeze_modules` parameter in the config. However, this does not mean that those parts of the model will act deterministically (e.g. due to dropout). In particular, for an ongoing experiment in the development of the diffusion model we are trying to overfit to a single encoded sample in latent space. This requires that at each iteration the frozen modules reliably output the same latent state. Hence we need a another parameter in the config that enforces not only freezing, but also sets the same modules into `eval()` mode. 

This might be useful more generally, but it should be discussed whether this is a desirable setting for other fine-tuning tasks.

Relates to #1379.

### Area

- [ ] datasets, data readers, data preparation and transfer
- [x] model
- [ ] science
- [ ] infrastructure and engineering
- [ ] evaluation, export and visualization
- [ ] documentation

**Comment by moritzhauschulz** [comment]
assign me
